aten/src/ATen/core/dispatch/DispatchKeyExtractor.h

```diff
@@ -98,15 +99,25 @@ namespace detail {
 struct CAFFE2_API DispatchKeyExtractor final {
 public:
   static DispatchKeyExtractor make(const FunctionSchema& schema) {
-    return DispatchKeyExtractor(schema.arguments().size());
+    TORCH_CHECK(schema.arguments().size() <= c10::utils::bitset::NUM_BITS,
+        "The function schema has ", schema.arguments().size(),
+        " arguments but this PyTorch build only supports ", c10::utils::bitset::NUM_BITS);
+    c10::utils::bitset dispatch_arg_indices_reverse;
+    for (size_t index = 0; index < schema.arguments().size(); ++index) {
+      if (schema.arguments()[index].type()->isSubtypeOf(TensorType::get()) || schema.arguments()[index].type()->isSubtypeOf(ListType::ofTensors())) {
+        dispatch_arg_indices_reverse.set(schema.arguments().size() - 1 - index);
+      }
+    }
+    return DispatchKeyExtractor(dispatch_arg_indices_reverse);
   }
 
   DispatchKey getDispatchKeyBoxed(DispatchKeySet backendsWithoutFallthrough, const torch::jit::Stack* stack) const {
     // TODO Unboxed dispatch supports TensorOptions (i.e. ScalarType/Device/Layout) arguments
     //      but boxed doesn't yet. See https://github.com/pytorch/pytorch/issues/26428
 
     DispatchKeySet ks;
-    for (const auto& ivalue : torch::jit::last(*stack, num_args_)) {
+    dispatch_arg_indices_reverse_.for_each_set_bit([&] (size_t reverse_arg_index) {
```

Comment:
did you check the assembly that all these fancy lambdas get inlined? (I still don't trust compilers :) )

Reply:
P126643185, looks inlined to me

Reply:
Yep, looks good as calls to isTensorList appear there nicely.